src/main/java/net/smartcosmos/edge/things/rest/ConflictOauth2ErrorHandler.java

```diff
@@ -0,0 +1,30 @@
+package net.smartcosmos.edge.things.rest;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.security.oauth2.client.http.OAuth2ErrorHandler;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.stereotype.Component;
+
+/**
+ * Error handler component that, in addition to the typical "No Error" HTTP response status codes, also accepts 409 Conflict as no error.
+ */
+@Component
+public class ConflictOauth2ErrorHandler extends OAuth2ErrorHandler {
+
+    @Autowired
+    public ConflictOauth2ErrorHandler(OAuth2ProtectedResourceDetails resource) {
+
+        super(resource);
+    }
+
+    @Override
+    public boolean hasError(ClientHttpResponse response) throws IOException {
+
+        HttpStatus status = response.getStatusCode();
+        return !(status.is2xxSuccessful() || status.is3xxRedirection() || status.is1xxInformational() || HttpStatus.CONFLICT.equals(status));
```

Comment:
Why not just 
`return super.hasError(response) && !HttpStatus.CONFLICT.equals(response.getStatusCode());`
Let the base class check for the error and just check for 409 if an error occurs...